### PR TITLE
[multistage][testing] Basic math func test and some more between test

### PR DESCRIPTION
**Pull request by 61yao**
1) Add basic math func test 
2) Add between literal test
3) Add between test for string type
4) Modified the test framework to support floating point fuzzy comparison

### CODE REVIEW

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+        "sql": "SELECT intCol %  10 FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "should return 0 for 0 value but we are not",
+        "description": "test mod on floating point literal with columns",
+        "sql": "SELECT intCol %  1.2 FROM {numTbl}"
+      },
+      {
+        "comment": "should round on the integer but we return a floating point",
```
Comment (agavra) [inline pinot-query-runtime/src/test/resources/queries/MathFuncs.json:276]: I think this comment was accidentally copied

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+        "description": "test mod on literals",
+        "sql": "SELECT 3 %  5 FROM {numTbl}"
+      },
+      {
+        "ignored": true,
+        "comment": "diff due to floating point comparison",
```
Comment (agavra) [inline pinot-query-runtime/src/test/resources/queries/MathFuncs.json:282]: can we fix the test framework to compare floating points properly? we can compare within .0001 error bound is probably fine

**Review by agavra — APPROVED** [review]
thanks! two minor comments

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json (hunk 2 above)
Comment (walterddr) [reply pinot-query-runtime/src/test/resources/queries/MathFuncs.json:282]: +1

File: pinot-query-runtime/src/test/resources/queries/Comparisons.json [hunk 3]
```diff
@@ -390,6 +390,7 @@
     "queries": [
       { "sql": "SELECT val BETWEEN small AND big FROM {tbl}" },
       { "sql": "SELECT val BETWEEN big AND small FROM {tbl}" },
+      { "sql": "SELECT val BETWEEN 2 AND 3 FROM {tbl}" },
```
Comment (walterddr) [inline pinot-query-runtime/src/test/resources/queries/Comparisons.json:393]: add `BETWEEN 3 AND 2`
literal values actually will be optimized out.

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json (hunk 2 above)
Comment (walterddr) [reply pinot-query-runtime/src/test/resources/queries/MathFuncs.json:282]: i think there's already a fix so let's see if these passes now

**Review by walterddr — COMMENTED** [review]
looks good overall. recommented some corner cases to test

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json (hunk 2 above)
Comment (61yao) [reply pinot-query-runtime/src/test/resources/queries/MathFuncs.json]: Done

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json (hunk 1 above)
Comment (61yao) [reply pinot-query-runtime/src/test/resources/queries/MathFuncs.json]: Removed

File: pinot-query-runtime/src/test/resources/queries/MathFuncs.json [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+  "multiply": {
+    "tables": {
+      "numTbl": {
+        "schema": [
+          {"name": "intCol", "type": "INT"},
+          {"name": "longCol", "type": "LONG"},
+          {"name": "doubleCol", "type": "DOUBLE"},
+          {"name": "floatCol", "type": "FLOAT"}
+        ],
+        "inputs": [
```
[the remark this answers is not in the document]
Comment (61yao) [reply pinot-query-runtime/src/test/resources/queries/MathFuncs.json:123]: I thought we are not supporting overflow. Added the test

File: pinot-query-runtime/src/test/resources/queries/Comparisons.json (hunk 3 above)
Comment (61yao) [reply pinot-query-runtime/src/test/resources/queries/Comparisons.json:393]: done

**Review by walterddr — APPROVED** [review]

**Pull request merged**